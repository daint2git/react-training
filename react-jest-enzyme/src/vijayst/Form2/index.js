import React from 'react'
import styled from 'styled-components'

const StyledForm = styled.form`
  padding: 20px;
`

const StyledInputText = styled.input`
  width: 200px;
  height: 40px;
`

const StyledButton = styled.button`
  color: white;
  backgroundcolor: black;
`

export default class Form2 extends React.Component {
  state = { email: '' }

  handleMailChange = e => {
    this.setState({ email: e.target.value })
  }

  handleAdd = e => {
    e.preventDefault()
    this.props.onAdd(this.state.email)
  }

  render() {
    return (
      <StyledForm>
        <StyledInputText
          type="text"
          name="email"
          value={this.state.value}
          onChange={this.handleMailChange}
        />
        <StyledButton onClick={this.handleAdd}>Add</StyledButton>
      </StyledForm>
    )
  }
}
