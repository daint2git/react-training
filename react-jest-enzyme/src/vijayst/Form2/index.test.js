import { mount, shallow, render } from 'enzyme'
import renderer from 'react-test-renderer'
import Form2 from './'

describe('Form2', () => {
  describe('Snapshot', () => {
    test('1. renderer', () => {
      const component = renderer.create(<Form2 />)
      const json = component.toJSON()
      expect(json).toMatchSnapshot()
    })

    test('2. shallow', () => {
      const wrapper = shallow(<Form2 />)
      expect(wrapper).toMatchSnapshot()
    })

    test('3. mount', () => {
      const wrapper = mount(<Form2 />)
      expect(wrapper).toMatchSnapshot()
    })

    test('4. render', () => {
      const wrapper = render(<Form2 />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
