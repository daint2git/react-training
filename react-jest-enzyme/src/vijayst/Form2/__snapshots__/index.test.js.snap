// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form2 Snapshot 1. renderer 1`] = `
<form
  className="sc-bdVaJa ksQzqc"
>
  <input
    className="sc-bwzfXH gmViiu"
    name="email"
    onChange={[Function]}
    type="text"
  />
  <button
    className="sc-htpNat keqZrr"
    onClick={[Function]}
  >
    Add
  </button>
</form>
`;

exports[`Form2 Snapshot 2. shallow 1`] = `
<styled.form>
  <styled.input
    name="email"
    onChange={[Function]}
    type="text"
  />
  <styled.button
    onClick={[Function]}
  >
    Add
  </styled.button>
</styled.form>
`;

exports[`Form2 Snapshot 3. mount 1`] = `
<styled.form>
  <form
    className="sc-bdVaJa ksQzqc"
  >
    <styled.input
      name="email"
      onChange={[Function]}
      type="text"
    >
      <input
        className="sc-bwzfXH gmViiu"
        name="email"
        onChange={[Function]}
        type="text"
      />
    </styled.input>
    <styled.button
      onClick={[Function]}
    >
      <button
        className="sc-htpNat keqZrr"
        onClick={[Function]}
      >
        Add
      </button>
    </styled.button>
  </form>
</styled.form>
`;

exports[`Form2 Snapshot 4. render 1`] = `
<form
  class="sc-bdVaJa ksQzqc"
>
  <input
    class="sc-bwzfXH gmViiu"
    name="email"
    type="text"
  />
  <button
    class="sc-htpNat keqZrr"
  >
    Add
  </button>
</form>
`;
